src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,67 @@
+package duke;
+
+/**
+ * Translates the user input string into a command object.
+ */
+public class Parser {
+
+    /**
+     * Takes in the user input and converts it into a command.
+     *
+     * @param input the user input string.
+     * @return the command object that represents the action that the user intends to make.
+     */
+    public static Command parse(String input) {
+        if (input.equals("bye")) return new ExitCommand();
+        if (input.equals("list")) return new ListCommand();
+        if (input.startsWith("done ")) return parseDone(input);
+        if (input.startsWith("delete ")) return parseDelete(input);
+        if (input.startsWith("find ")) return parseFind(input);
+        return comprehendTask(input);
+    }
+
+    private static DeleteCommand parseDelete(String input) {
+        int deletedTaskNumber;
+        try {
```

Comment:
Maybe you want to add some space here? I usually separate variables and logic with one empty line. (Whichever suits your style, as my suggestion is subjective)